### ISSUE TITLE
Add pre-built FIPS AMI

### ISSUE THREAD

**Issue by stevehipwell** [awslabs/amazon-eks-ami#1002]
**What would you like to be added**:
I'd like to be able to use a pre-built FIPS version of the AL2 EKS AMI provided by Amazon; AFAIK the spec for this is described in this [blog](https://aws.amazon.com/blogs/publicsector/enabling-fips-mode-amazon-linux-2/). I think this is required for both AMD64 & ARM64 but AMD64 support would do for now. I'd suggest the AMI name prefix of `amazon-eks-fips-node-`.

**Why is this needed**:
Everyone needing FIPS nodes needs to do the exact same thing which is the definition of toil and it should be easy to automate this as part of the AMI release process.

**Comment by seanorama** [comment]
+1 to this. I know of many who have built their own Packer pipelines for building FIPS into the image. Other Cloud providers have this by default or as an option.

**Comment by seanorama** [comment]
There is already the STIG option, so adding FIPS should be easy in comparison: https://aws.amazon.com/blogs/containers/building-stig-compliant-amis-for-amazon-eks/

Other Cloud providers have this by default in their k8s node images, or as a switch as part of their k8s service.

This is a major point of confusion and extra work, especially considering the blog post states that EKS meets FedRAMP High, which implies it has FIPS: https://aws.amazon.com/about-aws/whats-new/2021/04/amazon-eks-is-now-fedramp-high-compliant/

**Comment by stevehipwell** [comment]
I just spent most of the day Terraforming out the infrastructure to build an image for each EKS version for each architecture with EC2 ImageBuild. I think this is pretty much the definition of toil. And onboarding each account which needs to use the images will just add to that.

As an aside, the lack of an enable FIPS component for AL2 in ImageBuild seems like a major oversight. As does no support for source AMIs from SSM parameters.

**Comment by seanorama** [comment]
And very related: #898

**Comment by ghostsquad** [comment]
> I just spent most of the day Terraforming out the infrastructure to build an image for each EKS version for each architecture with EC2 ImageBuild. I think this is pretty much the definition of toil. And onboarding each account which needs to use the images will just add to that.

> As an aside, the lack of an enable FIPS component for AL2 in ImageBuild seems like a major oversight. As does no support for source AMIs from SSM parameters.

Would you be able to share what you've done to build these images?

**Comment by ghostsquad** [comment]
> I just spent most of the day Terraforming out the infrastructure to build an image for each EKS version for each architecture with EC2 ImageBuild. I think this is pretty much the definition of toil. And onboarding each account which needs to use the images will just add to that.

> As an aside, the lack of an enable FIPS component for AL2 in ImageBuild seems like a major oversight. As does no support for source AMIs from SSM parameters.

@stevehipwell Would you be able to share what you've done to build these images?

**Comment by WarheadsSE** [comment]
I have to second @stevehipwell here. Having these AMI available pre-made would be exponentially easier for customers wishing to operate with FIPS enabled.

**Comment by seanorama** [comment]
> Now that #1458 has been merged it'd be great to see FIPS images published by AWS so we don't have to manage custom image distribution across partitions.

This will be a huge benefit.

1. We maintain custom AMIs and then deal with the region and partition copying issues.
2. Environments typically require EBS encrypted to be enabled at the account/region level which further complicates copy/sharing as KMS is required.
3. This also gets EKS closer to being FedRAMP authorized by default, which it is not today.

**Comment by stevehipwell** [comment]
@cartermckinnon is this something which could be reevaluated in the context of AL2023?

**Comment by rajivml** [comment]
Is there a plan to support this, many of our customers are looking for this support from aws

**Comment by stevehipwell** [comment]
@bryantbiggs the Bottlerocket issue is relevant but it is not a replacement for this issue, Bottlerocket is great but not everyone is happy using it for all workloads. Also getting an AL2 FIPS AMI image should just be a packaging problem (I think the same goes for [AL2023](https://docs.aws.amazon.com/linux/al2023/ug/fips-mode.html)), while it looks like Bottlerocket is still waiting for certification (please correct me if I'm wrong and that has all been completed)?

So we're still waiting on a maintainer response. If we can't get a quick resolution for an AL2 image (which I suspect is the case) I'd be happy with working towards a pre-built AL2023 FIPS AMI to mirror Bottlerocket.

CC @cartermckinnon

**Comment by bryantbiggs** [comment]
> the Bottlerocket issue is relevant but it is not a replacement for this issue

I agree, and I am not saying its a replacement. But, the direction is still valid - if your organization requires FIPs support, the Bottlerocket AMI is going to be where you can achieve that out of the box. It is squarely within the 3 main goals of what Bottlerocket provides:

> Bottlerocket has three primary goals: [Minimal](https://bottlerocket.dev/#minimal) · [Safe Updates](https://bottlerocket.dev/#safe-updates) · [Security Focused](https://bottlerocket.dev/#security-focused).

This issue is still valid, but if you talk to any of us at AWS about meeting security and regulatory compliance and what level of "out of the box" support there is in EKS, the conversation will start with Bottlerocket

> Bottlerocket is great but not everyone is happy using it for all workloads

Our Bottlerocket team would love to learn more - please let your AWS account teams know and they can get a meeting setup to chat so that the Bottlerocket experience can be improved

**Comment by stevehipwell** [comment]
@bryantbiggs Bottlerocket covers most bases, and my recommendation to our users is that it should be their default. However issues like Bottlerocket being behind even AL2 on Containerd version ([1.6.x vs 1.7.x](https://github.com/bottlerocket-os/bottlerocket/issues/3876)) makes this conversation harder than it should be.

**Comment by rajivml** [comment]
@bryantbiggs are there any instructions on how to enable fips with bottlerocket Linux or is there a bottlerocket linux image with fips enabled available somewhere?

**Comment by bryantbiggs** [comment]
@rajivml please see https://github.com/bottlerocket-os/bottlerocket/issues/1667 for details on FIPs support with Bottlerocket

**Comment by jcorley-sysdig** [comment]
+1 would like to see pre-build fips al2023 amis available

**Comment by cazlo** [comment]
A FIPS Image Builder component would be nice valuable here, because if you are maintaining a system that cares about this Moderate/High Federal Government standard, you are probably also applying some level of image hardening customization (e.g. applying STIG hardening, CIS hardening, org specific security overrides such as system login prompt, etc.). 

There are existing Image Builder components for STIG hardening and CIS hardening.

I would also like to see pre-built images available so those who don't require the customization can spin up FIPS compliant EKS clusters in minutes instead of hours/days/weeks.
